### PR TITLE
Footnote context

### PR DESCRIPTION
**Pull request by jayvarner**

### CODE REVIEW

File: app/components/Footnote.tsx [hunk 1]
```diff
@@ -8,10 +8,17 @@ interface Props {
 }
 
 export default function Footnote({ number, children }: Props) {
-  const { backgroundColor, primaryTextColor } = useContext(ChapterContext);
+  const { backgroundColor, primaryTextColor, footnoteState, setFootnoteState, } = useContext(ChapterContext);
   const [isTextVisible, setIsTextVisible] = useState(false);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    setFootnoteState(footnoteState + 1);
```
Comment (NicholasLYang) [inline app/components/Footnote.tsx:16]: React setters can take the previous state:


```suggestion
    setFootnoteState(footnoteState => footnoteState + 1);
```

File: app/components/ScrollytellContent.tsx [hunk 2]
```diff
@@ -6,10 +6,11 @@ import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
 interface Props {
   children: ReactNodeLike;
   id: number;
-  update(): number;
+  setProgress: React.Dispatch<React.SetStateAction<number>>;
+  last: boolean;
 }
 
-export default function Content({ children, id, update }: Props) {
+export default function ScrollytellContent({ children, id, setProgress, last }: Props) {
```
Comment (NicholasLYang) [inline app/components/ScrollytellContent.tsx:13]: nit: I'd call this `isLast`

**Review by NicholasLYang — APPROVED** [review]

File: app/components/Footnote.tsx (hunk 1 above)
Comment (jayvarner) [reply app/components/Footnote.tsx:16]: Good to know.

File: app/components/ScrollytellContent.tsx (hunk 2 above)
Comment (jayvarner) [reply app/components/ScrollytellContent.tsx:13]: Fair

**Comment by jayvarner** [comment]
There are now too many conflicts with the footnote component. I'll start over and make another PR.

**Pull request closed without merging**